### ISSUE TITLE
Is it possible to use websocket for transport and noise for encryption?

### ISSUE THREAD

**Issue by fernvenue** [rapiz1/rathole#293]
Is it possible to use websocket for transport and noise for encryption?

_Originally posted by @fernvenue in https://github.com/rapiz1/rathole/issues/290#issuecomment-1742034945_

**Comment by rapiz1** [comment]
No. You have to change the code and somehow stack the transport layer,
which I think should be doable. But I wonder why not websocket with tls?

**Comment by fernvenue** [comment]
Hi @rapiz1, I just discussed this with @rucciva in https://github.com/rapiz1/rathole/pull/290. 

At start, I thought we might can use websocket to get better performance, but TLS is a little bit complex sometimes than noise, so I wondering that can we use websocket with noise :)

Then @rucciva suggests that I should do a benchmark, [here](https://github.com/rapiz1/rathole/pull/290#issuecomment-1742075248) I did, and the answer is that we should keep using TCP as protocol if we want best performance for both bandwidth and latency.

So, if anyone else have the same question, check the benchmark test, and I'm gonna close this issue for now.
